src/number_theory/legendre_symbol/add_character.lean

```diff
@@ -228,10 +228,19 @@ end
 of a field `R'`. It records which cyclotomic extension it is, the character, and the
 fact that the character is primitive. -/
 @[nolint has_nonempty_instance] -- can't prove that they always exist
-structure primitive_add_char (R : Type u) [comm_ring R] [fintype R] (R' : Type v) [field R'] :=
-(n : ℕ+)
-(char : add_char R (cyclotomic_field n R'))
-(prim : is_primitive char)
+def primitive_add_char (R : Type u) [comm_ring R] [fintype R] (R' : Type v) [field R'] :=
```

Comment:
Using `structure` caused a timeout, see https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/!4.234891.20.20FieldTheory.2ESplittingField/near/365719262 and https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/mysterious.20finsupp.20related.20timeout

Reply:
This comment, and link to zulip, should be in the source code.